config-templates/redhat/ocs-remote/4.6/README.md

```diff
@@ -0,0 +1,317 @@
+  ```
+  ibmcloud sat storage config create --name ocs-config --template-name ocs-local --template-version 4.7 -p "ocs-cluster-name=testocscluster" -p "mon-storage-class=vpc-custom-10iops-tier" -p "mon-size=50Gi" -p "osd-storage-class=vpc-custom-10iops-tier" -p "osd-size=150Gi" -p "num-of-osd=1" -p "worker-nodes=169.48.170.83,169.48.170.88,169.48.170.90" -p "ocs-upgrade=true" -p "ibm-cos-endpoint=https://s3.us-east.cloud-object-storage.appdomain.cloud" -p "ibm-cos-location=us-east-standard" -p "ibm-cos-access-key=xxx" -p "ibm-cos-secret-key=yyy"
+  ```
+
+4. Assign your configuration to your cluster groups.
+
+    ```
+    $ ibmcloud sat storage assignment create --name ocs-sub3 --group test-group2 --config ocs-config3
+    ```
+
+5. Verify that your configuration is updated.
+    ```
+    ic sat storage config get <config-name>
+    ```
+
+
+## Troubleshooting
+
+Check the status of your storage cluster.
+    ```
+    oc describe ocscluster
+    ```
+
+   Example output
+
+    ```
+    apiVersion: ocs.ibm.io/v1
+    kind: OcsCluster
+    metadata:
+      name: testocscluster
+    Spec:
+      Billing Type: hourly
+      Mon Size: 50Gi
+      Mon Storage Class Name: vpc-custom-10iops-tier
+      Num Of Osd: 2
+      Ocs Upgrade: false
+      Osd Size: 150Gi
+      Osd Storage Class Name: vpc-custom-10iops-tier
+      Status:
+      Storage Cluster Status:  Ready
+      workerNodes:
+      - 169.48.170.83
+      - 169.48.170.88
+      - 169.48.170.90
+    status:
+       storageClusterStatus: Ready
+    ```
+
+If the `storageClusterStatus` is `Progressing` or `Error`, the OCS installation has failed.
+
+### If your OCS installation fails, complete the following the steps to troubleshoot your deployment.
+1. Check the describe of storagecluster and cephcluster in the openshift-storage namespace and look at the `Events` and the `Status` sections
+
+    ```
+    oc describe storagecluster -n openshift-storage
+    oc describe cephcluster -n openshift-storage
+    ```
+
+2. You can use the toolbox available in rook community to debug OCS deploy issues. Run the following command to install the toolbox.
+    ```
+    oc patch ocsinitialization ocsinit -n openshift-storage --type json --patch  '[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]'
+    ```
+
+## Removing your OCS configuration
+
+1) Delete all the assignments and configurations created
+2) Remove the finalizer on the OcsCluster resource and delete it
+3) You'll have to first delete the `openshift-storage` namespace after removing finalizers on the resources under it
+4) Run the following command on each worker node to clean-up the files created by OCS.
+```
+oc debug node/<node name> -- chroot /host rm -rvf /var/lib/rook /mnt/local-storage
+```
```

Comment:
```suggestion
1. List your storage assignments and find the one that you used for your cluster. 
  ```sh
  ic sat storage assignment ls
  ```

2. Remove the assignment. After the assignment is removed, the driver pods and storage classes are removed from all clusters that were part of the storage assignment. 
  ```sh
  ic sat storage assignment rm --assignment <assignment_name>
  ```


3. Remove your storage assignment.
  ```sh
  ic sat storage assignment rm --assignment <assignment>
  ```


4. Clean up the remaining Kubernetes resources from your cluster. Save the following script in a file called `cleanup.sh` to your local machine.
    ```sh
    #!/bin/bash
    oc delete ns openshift-storage --wait=false
    sleep 20
    kubectl -n openshift-storage patch persistentvolumeclaim/db-noobaa-db-0 -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch cephblockpool.ceph.rook.io/ocs-storagecluster-cephblockpool -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch cephcluster.ceph.rook.io/ocs-storagecluster-cephcluster -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch cephfilesystem.ceph.rook.io/ocs-storagecluster-cephfilesystem -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch cephobjectstore.ceph.rook.io/ocs-storagecluster-cephobjectstore -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch cephobjectstoreuser.ceph.rook.io/noobaa-ceph-objectstore-user -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch cephobjectstoreuser.ceph.rook.io/ocs-storagecluster-cephobjectstoreuser -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch NooBaa/noobaa -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch backingstores.noobaa.io/noobaa-default-backing-store -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch bucketclasses.noobaa.io/noobaa-default-bucket-class -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n openshift-storage patch storagecluster.ocs.openshift.io/ocs-storagecluster -p '{"metadata":{"finalizers":[]}}' --type=merge
    sleep 20
    oc delete pods -n openshift-storage --all --force --grace-period=0
    oc delete ns local-storage --wait=false
    sleep 20
    kubectl -n local-storage patch localvolume.local.storage.openshift.io/local-block -p '{"metadata":{"finalizers":[]}}' --type=merge
    kubectl -n local-storage patch localvolume.local.storage.openshift.io/local-file -p '{"metadata":{"finalizers":[]}}' --type=merge
    sleep 20
    oc delete pods -n local-storage --all --force --grace-period=0
    ```

3. Run the `cleanup.sh` script.
  ```sh
  sh ./cleanup.sh
  ```

4. After you run the cleanup script, log in to each worker node and run the following commands.
  1. Deploy a debug pod and run `chroot /host`.
    ```
    oc debug node/<node name> -- chroot /host
    ```


  2. Run the following command to remove any files or directories on the specified paths.
    ```
    rm -rvf /var/lib/rook /mnt/local-storage
    ```

```
```

Reply:
Are the resources in the remote template deployed to the `local-storage` namespace? or is the namespace `remote-storage`?

Reply:
No, only the openshift-storage namespace is created for the remote template. I'll add only those lines of the script here